src/Devtools.tsx

```diff
@@ -16,14 +15,28 @@ import { Query, QueryCache, useQueryClient, onlineManager } from "@tanstack/soli
 import { getQueryStatusLabel, getQueryStatusColor } from "./utils";
 import { ArrowUp, ChevronDown, Offline, Search, Settings, Wifi } from "./icons";
 
+const [selectedStatus, setSelectedStatus] = createSignal<ReturnType<
+  typeof getQueryStatusLabel
+> | null>(null);
+const [selectedQueryHash, setSelectedQueryHash] = createSignal<string | null>(null);
+
 export const DevtoolsPanel: Component = () => {
   const styles = getStyles();
   const queryClient = useQueryClient();
   const queryCache = queryClient.getQueryCache();
 
   const queryCount = createSubscribeToQueryCache(queryCache, () => queryCache.getAll().length);
 
-  const queries = createMemo(on(queryCount, () => queryCache.getAll()));
+  const queries = createMemo(
+    on(
+      () => [queryCount(), selectedStatus()],
+      () => {
+        const curr = queryCache.getAll();
+        const status = selectedStatus();
+        return status === null ? curr : curr.filter((e) => getQueryStatusLabel(e) === status);
+      },
+    ),
+  );
```

Comment:
Thanks for working on this! I'll take a deeper look at this but there is a bug here. Try selecting `fetching` and then do a window refocus. You wont see the refetching queries in the loading state. Maybe the queryCount signal can change to the active selected query. Also maybe we should multiple selected statuses? So selectedStatus can be an array here

Reply:
fixed https://github.com/ardeora/solid-query-devtools/pull/2/commits/728e591f74299cec1909ad39570354ab377c840e